_posts/2016-09-22-approaching-tdd-outside-android-ii.md

```diff
@@ -0,0 +1,113 @@
+<li><b>ServiceLocator</b> - <a href="#servicelocator">As mentioned before</a>, this collaborator provides custom BankAccount instances to the Activity.</li>
+<li><b>Clock</b> - Time… time has a random factor and is outside of our control. To gain control over it, we have to encapsulate time randomness to have control over it during testing.</li></ul>
+
+We will see the collaborators in more detail later on, while we implement them.
+
+Once we have finished writing the acceptance test and we see it <b>failing for the right reason</b> (The view is not showing the statement), we will dive into the inner loop and start writing the individual pieces required for the feature.
+
+<img src="http://panavtec.me/wp-content/uploads/2016/08/acc2.png" alt="Outside-in acceptance test coverage" width="831" height="271" class="aligncenter size-full wp-image-1785" />
+
+<h2>Inner loop - Unit test</h2>
+
+The first collaborator to implement is the one in the outer layer of the system, in this case, the <b>ShowStatementActivity</b>.
+
+<ul style="display: inline-block; list-style: none; text-align: left;">
+<li><span style="color: #d32f2f; padding-right: 5px;">●</span><a href="https://github.com/CarlosMChica/AndroidBankKata/commit/8d6122f71067f9d8c12f70f3fbdade2a6b707d58">c22ab01</a> - Let's start writing a failing test that will lead us to create the UI to show the statement. Here we are going to use a vertical RecyclerView.</li>
+<li><span style="color: #388e3c; padding-right: 5px;">●</span><a href="https://github.com/CarlosMChica/AndroidBankKata/commit/abc15e855edd891a7d6e1c98e1a2849622898173"> 8d6122f</a> - Once we have the failing test, we have to make it pass by loading the activity layout, holding a reference to the RecyclerView and setting a LinearLayoutManager with a vertical orientation.</li>
+<li><span style="color: #1976d2; padding-right: 5px;">●</span><a href="https://github.com/CarlosMChica/AndroidBankKata/commit/08a5937a7ceced76a527f399e2597f4ea916634e">08a593</a> - We started by including all the code inside the onCreate method. In the refactor phase, we decided to extract a <i>initView</i>  method to wrap view initialisation code.</li></ul>
+
+At this point, the RecyclerView is initialised and ready to hold the statement rows, but we still need to <b>show the account statement</b>. Let's do it.
+
+<ul style="display: inline-block; list-style: none; text-align: left;">
+<li><span style="color: #d32f2f; padding-right: 5px;">●</span><a href="https://github.com/CarlosMChica/AndroidBankKata/commit/8520fbba9b3ed7e4b330544c23eb1027db77f897">8520fb</a> - The test has to verify that when the activity is launched, the view gets attached to the account and the BankAccount <i>showStatement</i> method is called. As mentioned before, we do not have control over the activity instantiation so we need to use the ServiceLocator to provide a test BankAccount for the activity.</li>
```

Comment:
more italics instead of tics... I suggest going through the whole text and looking for instances